### DISCUSSION TITLE
Metadata CreationTimestamp NULL for Generated CRDs

### DISCUSSION THREAD

**Discussion by shaneutt** [kubernetes-sigs/kubebuilder#2315] [Support] [answered]
I'm finding when using the `Makefile` that `kubebuilder` `v3.1.0` provides for new projects that all the CRDs generated by it come with a `creationTimestamp: null` field in the metadata. I just figured I would check in here to ask if anyone knew if there's an option to that can be set on the types, or maybe options for `controller-gen` that would stop these from being emitted?

**Comment by camilamacedo86** [comment, marked as answer]
Hi @shaneutt,

That is true. Controller-gen is creating it as null: https://github.com/kubernetes-sigs/kubebuilder/blob/26d12ab1134964dbbc3f68877ebe9cf6314e926a/testdata/project-v3/config/crd/bases/crew.testproject.org_admirales.yaml#L2-L8

Could you please raise an issue against controller-gen: https://github.com/kubernetes-sigs/controller-tools? 
Also, feel free to contribute. Note that your collaboration is very welcome 🥇

**Reply by WindowsXp-Beta** [reply]
A lot of search finally leads me here.
Maybe you should point this out in the [tutorial](https://book.kubebuilder.io/cronjob-tutorial/running.html)? Otherwise `make install` would run into
```
error: unable to decode "STDIN": parsing time "null" as "2006-01-02T15:04:05Z07:00": cannot parse "null" as "2006"
```
Btw, I found a related [issue](https://github.com/kubernetes-sigs/controller-tools/issues/402) which is not resolved but is closed.

**Reply by pier-oliviert** [reply]
Bumping controller-gen version fixes the issue, here's an example:

https://github.com/kubernetes-sigs/kueue/pull/826/files

**Reply by shaneutt** [reply]
Thanks!

**Comment by camilamacedo86** [comment]
Closing since it is answered.
